### PR TITLE
IBP-6119-implement-new-manage-studies-landing-page-deprecate-browse-studies

### PR DESCRIPTION
**Pull request by lik-83**

### CODE REVIEW

File: src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java [hunk 1]
```diff
@@ -295,6 +295,7 @@ public List<DatasetDTO> getDatasets(final Integer studyId, final Set<Integer> da
 		final List<DatasetDTO> datasetDTOs = new ArrayList<>();
 		for (final org.generationcp.middleware.domain.dms.DatasetDTO datasetDTO : datasetDTOS) {
 			final DatasetDTO datasetDto = mapper.map(datasetDTO, DatasetDTO.class);
+			// TODO: should also rename environment dataset?
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java:298]: @lik-83 anything we should do with this TODO?

File: src/main/java/org/ibp/api/java/impl/middleware/study/StudyTreeServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+		return this.studyTreeService.updateStudyTreeFolder(parentId, newFolderName);
+	}
+
+	@Override
+	public void deleteStudyFolder(final String cropName, final String programUUID, final Integer folderId) {
+		this.studyTreeValidator.validateFolderId(folderId, programUUID);
+		this.validateProgram(cropName, programUUID);
+		this.studyTreeValidator.validateFolderHasNoChildren(folderId, "study.folder.delete.has.child", programUUID);
+
+		// TODO: should only the owner be able to delete the folder?
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/study/StudyTreeServiceImpl.java:74]: @lik-83 not sure how it was working in the previous module

File: src/main/java/org/ibp/api/java/study/StudyService.java [hunk 3]
```diff
@@ -25,17 +25,24 @@ public interface StudyService {
...
+	@Deprecated
 	long countFilteredStudies(String programUUID, StudySearchRequest studySearchRequest);
 
 	void deleteNameTypeFromStudies(Integer nameTypeId);
+
+	List<StudySearchResponse> searchStudies(String programUUID, StudySearchRequest studySearchRequest, Pageable pageable);
+
+	long countSearchStudies(String programUUID, StudySearchRequest studySearchRequest);
+
+	StudyDetailsDTO getStudyDetails(final String programUUID, Integer studyId);
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/study/StudyService.java:46]: @lik-83 remove final

File: src/main/java/org/ibp/api/rest/dataset/DatasetResource.java [hunk 4]
```diff
@@ -93,7 +93,7 @@ public class DatasetResource {
 
 	@ApiOperation(value = "Get Dataset Columns", notes = "Retrieves ALL MeasurementVariables (columns) associated to the dataset, "
 		+ "that will be shown in the Observation Table")
-	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES', 'MANAGE_STUDIES', 'BROWSE_STUDIES')" + PermissionsEnum.HAS_MANAGE_STUDIES_VIEW)
+	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES', 'MANAGE_STUDIES', 'VIEW_STUDIES')" + PermissionsEnum.HAS_MANAGE_STUDIES_VIEW)
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/dataset/DatasetResource.java:96]: @cuenyad this comment applies to all this class
should we add VIEW_STUDIES to HAS_MANAGE_STUDIES_VIEW and simplify most of the PreAuthorities in this resource? 
Not that view studies is below manage studies, makes sense to me to add it to the enum and simplify the list or authorities when this is used:
``
hasAnyAuthority('ADMIN','STUDIES', 'MANAGE_STUDIES', 'VIEW_STUDIES')" + PermissionsEnum.HAS_MANAGE_STUDIES_VIEW
``

File: src/main/java/org/ibp/api/rest/derived/DerivedVariableResource.java [hunk 5]
```diff
@@ -86,7 +86,7 @@ public ResponseEntity<Set<FormulaVariable>> formulaVariables(
 	}
 
 	@ApiOperation(value = "Count Calculated Traits", notes = "Count the calculated traits (derived traits) in a specified dataset(s)")
-	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES', 'BROWSE_STUDIES')" + PermissionsEnum.HAS_MANAGE_STUDIES_VIEW)
+	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES', 'VIEW_STUDIES')" + PermissionsEnum.HAS_MANAGE_STUDIES_VIEW)
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/derived/DerivedVariableResource.java:89]: @cuenyad same comment in relation to the enum here

File: src/main/java/org/ibp/api/rest/study/GermplasmStudySourceResource.java [hunk 6]
```diff
@@ -42,7 +42,7 @@ public class GermplasmStudySourceResource {
 
 	@ApiOperation(value = "It will retrieve all generated germplasm in a study",
 		notes = "It will retrieve all generated germplasm in a study")
-	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES', 'BROWSE_STUDIES')"
+	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES', 'VIEW_STUDIES')"
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/study/GermplasmStudySourceResource.java:45]: @cuenyad same here about the enu,

File: src/main/java/org/ibp/api/rest/study/GermplasmStudySourceResource.java [hunk 7]
```diff
@@ -70,7 +70,7 @@ public ResponseEntity<List<GermplasmStudySourceDto>> getGermplasmStudySourceTabl
 
 	@ApiOperation(value = "Post generated germplasm in a study search", notes = "Post generated germplasm in a study search")
 	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/germplasm-sources/search", method = RequestMethod.POST)
-	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES', 'BROWSE_STUDIES')"
+	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES', 'VIEW_STUDIES')"
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/study/GermplasmStudySourceResource.java:73]: @cuenyad same here about the enum

File: src/main/java/org/ibp/api/rest/study/StudyInstanceResource.java [hunk 8]
```diff
@@ -57,7 +57,7 @@ public ResponseEntity<Void> deleteStudyInstances(final @PathVariable String crop
 
 	@ApiOperation(value = "List all study instances with basic metadata.",
 		notes = "Returns list of all study instances with basic metadata.")
-	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES','BROWSE_STUDIES')"
+	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES','VIEW_STUDIES')"
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/study/StudyInstanceResource.java:60]: @cuenyad same here with the enum

File: src/main/java/org/ibp/api/rest/study/StudyInstanceResource.java [hunk 9]
```diff
@@ -75,7 +75,7 @@ public ResponseEntity<List<StudyInstance>> listStudyInstances(final @PathVariabl
 
 	@ApiOperation(value = "Get study instance with basic metadata.",
 		notes = "Get study instances with basic metadata.")
-	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES', 'BROWSE_STUDIES')" + PermissionsEnum.HAS_MANAGE_STUDIES_VIEW)
+	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES', 'VIEW_STUDIES')" + PermissionsEnum.HAS_MANAGE_STUDIES_VIEW)
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/study/StudyInstanceResource.java:78]: @cuenyad same here with the enum

File: src/main/java/org/ibp/api/rest/dataset/DatasetResource.java (hunk 4 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/rest/dataset/DatasetResource.java:96]: For now, I prefer to keep View Studies separate from HAS_MANAGE_STUDIES_VIEW, to maintain the same permissions as defined in MANAGE_STUDIES_VIEW_PERMISSIONS from permissions.ts. When we start with the granular permissions in the study, the HAS_MANAGE_STUDIES_VIEW will not make sense anymore.

File: src/main/java/org/ibp/api/rest/study/GermplasmStudySourceResource.java (hunk 6 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/rest/study/GermplasmStudySourceResource.java]: done!

File: src/main/java/org/ibp/api/rest/derived/DerivedVariableResource.java (hunk 5 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/rest/derived/DerivedVariableResource.java]: done!

File: src/main/java/org/ibp/api/rest/study/StudyInstanceResource.java (hunk 8 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/rest/study/StudyInstanceResource.java]: done!

File: src/main/java/org/ibp/api/rest/study/StudyInstanceResource.java (hunk 9 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/rest/study/StudyInstanceResource.java]: done!

File: src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java (hunk 1 above)
Comment (lik-83) [reply src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java:298]: I removed this TODO.

File: src/main/java/org/ibp/api/java/impl/middleware/study/StudyTreeServiceImpl.java (hunk 2 above)
Comment (lik-83) [reply src/main/java/org/ibp/api/java/impl/middleware/study/StudyTreeServiceImpl.java:74]: Currently, any user can delete any folder

File: src/main/java/org/ibp/api/java/study/StudyService.java (hunk 3 above)
Comment (lik-83) [reply src/main/java/org/ibp/api/java/study/StudyService.java:46]: Done!

**Pull request merged**